File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> m;
        int prefix = 0;
        int count = 0;
        for(int i=0;i<nums.size();i++){
            prefix += nums[i];
            if(prefix == k) count++;
            if(m[(prefix-k)]) count += m[(prefix-k)];
            m[prefix] += 1;
            
        }
        return count;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $k
     * @return Integer
     */
    function subarraySum($nums, $k) {
        $sum = 0;
        $map = [];
        $count = 0;

        foreach($nums as $num){
            $sum += $num;
            $target = $sum-$k;
            if(isset($map[$target])) $count += $map[$target];
            if($target == 0) $count++;
            $map[$sum] = isset($map[$sum])?$map[$sum]+1:1;
        }

        return $count;
        
    }
}